app/models/workflow_step.rb

```diff
@@ -23,6 +23,10 @@ class WorkflowStep < ApplicationRecord
 
   scope :for_version, ->(version) { where(version:) }
 
+  # On delete: raises an exception if there are is any workflow metadata
+  # Don't destroy because there may be other steps that reference the same druid
+  has_one :workflow_metadata, foreign_key: 'druid', primary_key: 'druid', dependent: :restrict_with_error, inverse_of: :workflow_step
+
```

Comment:
hmmm...this may be a problem actually.... since when you create a new workflow, we currently destroy any instances of the current one